packages/block-library/src/page-list/block.json

```diff
@@ -0,0 +1,20 @@
+{
+	"apiVersion": 2,
+	"name": "core/page-list",
+	"category": "widgets",
+	"usesContext": [
+		"textColor",
+		"customTextColor",
+		"backgroundColor",
+		"customBackgroundColor",
+		"fontSize",
+		"customFontSize",
+		"showSubmenuIcon"
```

Comment:
It feels like "context" is being abused in the navigation block, what happens if I use this block independently and I want to customize these things?

Reply:
Looked a bit at this. It relates to the block support mechanism and parent/child relationship.

Braindump of what we have now:

- a block wants to serialize styles to its wrapper => default supports mechanism
- a block wants to serialize styles to any other node => support mechanism with skip serialization (but only for its own markup, can't affect children)
- a block wants to serialize styles to children blocks or or a children wants to take styles from the parent => block context

One thing we could explore would be to allow blocks to automatically expose some block supports via context without having to declare context support explicitely. Was that what you had in mind, Miguel?

Reply:
I added the context in to match the Navigation Link block, so that all the links in the Navigation could be styled equally. When Page List is used as an independent block, those styling options aren't available. This seemed ok as the standalone block reproduces the functionality of the Pages widget, and similar blocks like Latest Posts also have limited styling.
As for the styling mechanism used I think we should go with the same strategy for both Navigation Link and Page List so that they work consistently inside the Navigation block, but I'm not super knowledgeable about how Global Styles work so not sure what it might look like!

Reply:
> One thing we could explore would be to allow blocks to automatically expose some block supports via context without having to declare context support explicitely. Was that what you had in mind, Miguel?

Actually, I'm much less concerned about explicitly including Global Styles / Block Supports keys under `usesContext`, than I am about manually replicating some of `gutenberg_apply_colors_support` as part of `render_block_core_page_list` (see `block_core_page_list_build_css_colors`). This is where I think it'd be best to see where, if anywhere, GS/BS could improve to accommodate scenarios like Page List.

Reply:
In playing with this code I've noticed the block attributes and the block context have the same shape, so we can apply the same transformations on that data to create classes and styles ― it's a matter of finding how.

A small step in making both cases more similar could be https://github.com/WordPress/gutenberg/pull/32807 (experimented only with colors to see how it feels).